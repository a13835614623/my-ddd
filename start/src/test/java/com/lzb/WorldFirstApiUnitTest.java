package com.lzb;

import java.time.OffsetDateTime;
import java.time.ZoneOffset;

import cn.hutool.http.HttpRequest;
import cn.hutool.http.HttpUtil;
import org.jetbrains.annotations.NotNull;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import static com.lzb.WorldFirstApi.*;

/**
 * 阿里万里汇对接<br/>
 * Created on : 2023-09-24 13:59
 * @author lizebin
 */
class WorldFirstApiUnitTest extends BaseUnitTest {

    @Test
    @DisplayName("构造header")
    void should_constructor_header() throws Exception {

        Header header = Header.builder()
                .signature(getSignature())
                .requestTime(OffsetDateTime.of(2023, 1, 2, 2, 2, 2, 2, ZoneOffset.ofHours(8)))
                .build();
        assertJSON(header.formatToMap());
    }

    private static Signature getSignature() {
        return Signature.builder()
                .httpMethod("POST")
                .uri("/amsin/api/v1/business/account/inquiryBalance")
                .requestBody("{\"transferFactor\":{\"transferFundType\":\"GLOBAL_WORLDFIRST\"},\"currency\":\"USD\"}")
                .build();
    }

    @Test
    @DisplayName("构造签名")
    void should_build_signature() throws Exception {
        String signature = Signature.builder()
                .httpMethod("POST")
                .uri("/amsin/api/v1/business/account/inquiryBalance")
                .requestBody("{\"transferFactor\":{\"transferFundType\":\"GLOBAL_WORLDFIRST\"},\"currency\":\"USD\"}")
                .build()
                .sign("2022-03-02T15:03:30+08:00");
        assertThat("F%2FOWHMzXqh8519ckwpAHrUDSrV2h%2B7ErKne0yL8Pg03TooyMDHotfcSsJzKBs3p1SSIZ7Ey1AkC8p%2FxFL3n51EkTPbwp"
                + "La5gAyPBPsZg9imNt1lFfJFgA4%2FnkSfBQuEwBkKZQCQTyDsn3Pkm1mU6N4TRyQzTt9RKabnm2AYRlP%2BjLL8Nbzh%2Bap0VsMcvt0qQyJ"
                + "2YhMVENlKw9Mxv5zXLca1opv5sE9wwPdB0SI5Sv3lJKV1c90T8L4AHOQwWg2tp%2BbRJL6PHqXosnwzezDTSdrAW2kV9x8wtc%2BRIcGNnDGQ"
                + "SPGBLlVpqHGg20%2FGJEO5CD65B0KNaNB92oAlk%2FgH2qw%3D%3D").isEqualTo(signature);
    }

}
