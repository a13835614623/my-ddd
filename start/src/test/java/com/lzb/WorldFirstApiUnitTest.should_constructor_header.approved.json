{
  "ClientId" : "Client-Id:3J5Y8Y382Y3K3105035",
  "Content-Type" : "Content-Type:application/json",
  "Request-Time" : "Request-Time:2023-01-02T02:02:02+08:00",
  "Signature" : "Signature:algorithm=RSA256, keyVersion=2, signature=BWTIdphO8jo1krBja4V6PMcd6XMrzmYDgOjKfcmYaK9tjFMktOH8alFOKADWcbNzeppSu%2BrEn76aKJZSVcsffq6f9tlMIGweaG7Jeeb%2FmvTXROGAaqd55ZMxxtqUnUzihDasAqL%2FegaUone04sJOlOZT8U5ia5zw94ilM%2Fwif%2BBTBlhhT5%2B5n7DqQ6QkJNaDxS3yvs2%2FdqbW4ndyrr%2BxrxnRTBd9JqyXStKSLV4St3A67Kk9grR1fJE4Ft6kW%2BrVHxI7Sr8ewgeBAyVz3OL1NMpgUHygIjRKhffO6p5aY%2BWDcc9DbXwEE8NgmY91aVQOU2JJgm%2Fyf7dYdOufDs6l4g%3D%3D"
}